src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -991,13 +997,22 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 									if (stringParam is KnownStringValue stringValue) {
 										switch (fieldPropertyOrEvent) {
 										case IntrinsicId.Type_GetEvent:
-											MarkEventsOnTypeHierarchy (ref reflectionContext, systemTypeValue.TypeRepresented, filter: e => e.Name == stringValue.Contents, bindingFlags);
+											if ((bindingFlags & BindingFlags.IgnoreCase) == BindingFlags.IgnoreCase || (int) bindingFlags > 255)
+												RequireDynamicallyAccessedMembers (ref reflectionContext, DynamicallyAccessedMemberTypes.PublicEvents | DynamicallyAccessedMemberTypes.NonPublicEvents, value, calledMethodDefinition);
+											else
+												MarkEventsOnTypeHierarchy (ref reflectionContext, systemTypeValue.TypeRepresented, filter: e => e.Name == stringValue.Contents, bindingFlags);
 											break;
```

Comment:
Nit: Seeing how this is repeated I wonder if it would make sense to "hide" the binding flags check inside the Mark* methods... Not a big deal though, we could also leave it as is if you like this better.

Reply:
I think I like this way because is visible along with the other fallbacks